components/ui/SelectStore.tsx

```diff
@@ -0,0 +1,170 @@
+        index,
+      ) => (
+        <li id={String(index)}>
+          <button
+            onClick={() => onClickFunction(index)}
+          >
+            {name.toUpperCase()}
+          </button>
+        </li>
+      ))}
+    </>
+  );
+}
+
+function ModalItems(
+  { option, isStoreDetailsOpen, onCloseFunction }: ModalItemsProps,
+) {
+  return (
+    <>
+      {option && (
+        <Modal
+          loading="lazy"
+          title={option.name.toUpperCase()}
+          mode="center"
+          open={isStoreDetailsOpen}
+          onClose={() => onCloseFunction()}
+        >
+          <div class="container flex">
+            <div
+              href="#"
+              class="flex flex-col gap-2 lg:w-[196px] lg:h-[90px] self-center"
+            >
+              <figure class="self-center mix-blend-multiply">
+                <Image
+                  class="card h-full"
+                  src={option.image}
+                  alt={option.name ||
+                    option.type ||
+                    option.localization}
+                  width={105}
+                  height={90}
+                  loading="eager"
+                />
+              </figure>
+            </div>
+            <div class="flex-column pl-4">
+              <p>{option.type}</p>
+              <p>{option.localization}</p>
+
+              {option?.phone && (
+                <p>
+                  Telefone: <a class="font-light">{option.phone}</a>
+                </p>
+              )}
+              {option?.phone && (
+                <p>
+                  WhatsApp: <a class="font-light">{option.whatsapp}</a>
+                </p>
+              )}
+            </div>
+          </div>
+          <button
+            class="flex justify-end"
+            onClick={() => onCloseFunction()}
+          >
+            <p class="border border-base-200 rounded py-2 px-4">Fechar</p>
+          </button>
+        </Modal>
+      )}
+    </>
+  );
+}
+
+export default function SelectStore(props: Props) {
+  const {
+    selectLabel = "",
+    options = [
+      {
+        name: "Loja 1",
+        type: "Loja teste",
+        localization: "Piso teste",
+        image:
+          "https://ik.imagekit.io/decocx/tr:w-680,h-680/https:/ozksgdmyrqcxcwhnbepg.supabase.co/storage/v1/object/public/assets/239/fdcb3c8f-d629-485e-bf70-8060bd8a9f65",
+        phone: "(21) 11111-1111",
+      },
+    ],
+  } = props;
+  const isStoreDetailsOpen = useSignal(false);
+  const indexStore = useSignal(0);
+  const slicedOptions = useSignal(options);
+  const inputValue = useSignal("");
+
+  const searchbarItemsFunction = (index: number) => {
+    indexStore.value = index;
+    isStoreDetailsOpen.value = true;
+  };
+  const modalItemsFunction = () => {
+    isStoreDetailsOpen.value = false;
+  };
+
+  useEffect(() => {
+    slicedOptions.value = options.filter((item) =>
+      item.name.toUpperCase().includes(inputValue.value.toUpperCase())
+    );
+  }, [inputValue.value]);
```

Comment:
Voce nao precisa de um signal e um useEffect para isso, basta voce colocar a variavel em tempo de execuçao pois a unica coisa que precisa fazer é exibi-la, guardar ela em um state/signal é dispensavel.
```suggestion
    const slicedOptions = options.filter((item) =>
      item.name.toUpperCase().includes(inputValue.value.toUpperCase())
    );
```